diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fec260..c6f0d8d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,7 @@ cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
 ##      -DGTK_USE_NATIVEBOOK=1|0                   // Under GTK, use wxNotebook instead of wxAuiNotebook. Default is set to 0                                         #
 ##      -DWITH_WXPATH=<fullpath>                   // Specify a particular wxWidgets build to use. The format must be /path/to/different_wx-config/directory/         #
 ##      -DMAKE_DEB=1|0                             // When set to 1, you can use make package to create .deb file for codelite                                        #
+##      -DPLUGINS=<plugin1>;<plugin2>...           // Specify semi-colon separated user plugin list                                                                   #
 #######################################################################################################################################################################
 
 
@@ -383,6 +384,15 @@ if ( NOT NO_CPP_PLUGINS )
         message("-- Adding MacBundler...")
         add_subdirectory(MacBundler)
     endif ( APPLE )
+
+    if ( PLUGINS )
+        FOREACH ( PLUGIN ${PLUGINS} )
+            message("-- User plugin: ${PLUGIN}")
+            add_subdirectory(${PLUGIN})
+        ENDFOREACH ( PLUGIN )
+	unset(PLUGINS CACHE)
+    endif ( PLUGINS )
+
 endif ( NOT NO_CPP_PLUGINS )
 
 add_subdirectory(Tweaks)
